### PR TITLE
db interface + fakeredis for tests

### PR DESCRIPTION
**Pull request by jadbox**

### CODE REVIEW

File: test/layer2libTest.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const web3 = new Web3()
 const Layer2lib = require('../src/index.js')
 const Promise = require('bluebird');
 
-const redis = require('redis')
```
Comment (nginnever) [inline test/layer2libTest.js]: Can you comment this out rather than delete as I will need to have this for my tests

File: test/layer2libTest.js [hunk 2]
```diff
@@ -15,20 +15,21 @@ async function test() {
 
   // ALICE
   let optionsAlice = {
-    provider: 'http://localhost:8545',
```
Comment (nginnever) [inline test/layer2libTest.js]: Is there a reason to pull the provider url for web3 out of the options?

Comment (nginnever) [reply test/layer2libTest.js]: If so this would have to be done for bob's instance and below as well

Comment (jadbox) [reply test/layer2libTest.js]: @nginnever it's just idiomatic Node to have your primary connection url as an isolated and first param to your constructor.

Comment (jadbox) [reply test/layer2libTest.js]: fixed bob's url

File: test/layer2libTest.js (hunk 1 above)
Comment (jadbox) [reply test/layer2libTest.js]: done!

File: test/layer2libTest.js (hunk 2 above)
Comment (nginnever) [reply test/layer2libTest.js]: sounds good

**Pull request merged**